src/components/UsernameInput/UsernameInput.tsx

```diff
@@ -21,6 +27,9 @@ const UsernameInput: FC<UsernameInputProps> = ({ state, onInputChange, label })
 
     const handleInputChange = (value: string) => {
         onInputChange({ variant: state.variant, textValue: value });
+        if (state.variant === "email") {
+            validateEmail(value);
+        }
```

Comment:
Great! Now we have to think about the edge cases 

for example 
- If I start typing and decide that I no longer want an email, the email validation yells at me, can we make it such that if the input is empty, we set **emailValidation** to `undefined` ? 🤔 
<img height="200" alt="image" src="https://github.com/evargast/signature-generator/assets/52969178/a5cd9b6c-3b95-40a1-8e30-1a0e3869b8be">

- If I enter an invalid email, while the validation does tell me that it is invalid, we are updating the provider state and displaying the invalid email within the signature, is there a way to only update the email state if it is valid? 👀 
<img width="400" alt="image" src="https://github.com/evargast/signature-generator/assets/52969178/281dc6a5-39b3-44e4-8757-6e6eca26ae96">